libcloud/test/compute/test_retry_limit.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import socket
+import tempfile
+from libcloud.utils.py3 import httplib
+
+from mock import Mock, patch, MagicMock
+
+from libcloud.common.base import Connection
+from libcloud.common.base import Response
+from libcloud.common.exceptions import RateLimit
+from libcloud.test import unittest
+
+CONFLICT_RESPONSE_STATUS = [
+    ('status', '429'), ('reason', 'CONFLICT'),
+    ('retry_after', '3'),
+    ('content-type', 'application/json')]
+SIMPLE_RESPONSE_STATUS = ('HTTP/1.1', 429, 'CONFLICT')
+
+
+class RateLimitTestCase(unittest.TestCase):
+
+    def _raise_socket_error(self):
+        raise socket.gaierror('')
+
+    def test_retry_connection(self):
```

Comment:
@Kami : Yes this test doesn't make any sense so I'll remove this in next PR. Also I noticed this [test](https://github.com/apache/libcloud/blob/trunk/libcloud/test/test_connection.py#L285) actually testing retry logic along with two other test below it, I'm thinking no other test are required to test it. Let me know your thoughts.